subsys/fb/cfb_shell.c

```diff
@@ -0,0 +1,470 @@
+		shell_error(shell, HELP_INIT);
+		return -ENODEV;
+	}
+
+	shell_print(shell, "param: %s=%d", param_name[CFB_DISPLAY_WIDTH],
+		    cfb_get_display_parameter(dev, CFB_DISPLAY_WIDTH));
+
+	return 0;
+}
+
+static int cmd_get_param_ppt(const struct shell *shell, size_t argc,
+			     char *argv[])
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	if (!dev) {
+		shell_error(shell, HELP_INIT);
+		return -ENODEV;
+	}
+
+	shell_print(shell, "param: %s=%d", param_name[CFB_DISPLAY_PPT],
+		    cfb_get_display_parameter(dev, CFB_DISPLAY_PPT));
+
+	return 0;
+}
+
+static int cmd_get_param_rows(const struct shell *shell, size_t argc,
+			     char *argv[])
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	if (!dev) {
+		shell_error(shell, HELP_INIT);
+		return -ENODEV;
+	}
+
+	shell_print(shell, "param: %s=%d", param_name[CFB_DISPLAY_ROWS],
+		    cfb_get_display_parameter(dev, CFB_DISPLAY_ROWS));
+
+	return 0;
+}
+
+static int cmd_get_param_cols(const struct shell *shell, size_t argc,
+			     char *argv[])
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	if (!dev) {
+		shell_error(shell, HELP_INIT);
+		return -ENODEV;
+	}
+
+	shell_print(shell, "param: %s=%d", param_name[CFB_DISPLAY_COLS],
+		    cfb_get_display_parameter(dev, CFB_DISPLAY_COLS));
+
+	return 0;
+}
+
+static int cmd_init(const struct shell *shell, size_t argc, char *argv[])
+{
+	int err;
+
+	dev = device_get_binding(DISPLAY_DRIVER);
+
+	if (dev == NULL) {
+		shell_error(shell, "Device not found");
+		return -ENODEV;
+	}
+
+	err = cfb_framebuffer_init(dev);
+	if (err) {
+		shell_error(shell, "Framebuffer initialization failed!");
+		return err;
+	}
+
+	shell_print(shell, "Framebuffer initialized: %s", DISPLAY_DRIVER);
+
+	cmd_clear(shell, argc, argv);
+
+	return err;
+}
+
+SHELL_CREATE_STATIC_SUBCMD_SET(sub_cmd_get_param) {
+
+	SHELL_CMD_ARG(all, NULL, NULL, cmd_get_param_all, 1, 0),
+	SHELL_CMD_ARG(height, NULL, NULL, cmd_get_param_height, 1, 0),
+	SHELL_CMD_ARG(width, NULL, NULL, cmd_get_param_width, 1, 0),
+	SHELL_CMD_ARG(ppt, NULL, NULL, cmd_get_param_ppt, 1, 0),
+	SHELL_CMD_ARG(rows, NULL, NULL, cmd_get_param_rows, 1, 0),
+	SHELL_CMD_ARG(cols, NULL, NULL, cmd_get_param_cols, 1, 0),
+	SHELL_SUBCMD_SET_END
+};
+
+SHELL_CREATE_STATIC_SUBCMD_SET(sub_cmd_scroll) {
+
+	SHELL_CMD_ARG(vertical, NULL, HELP_PRINT, cmd_scroll_vert, 4, 0),
+	SHELL_CMD_ARG(horizontal, NULL, HELP_PRINT, cmd_scroll_horz, 4, 0),
+	SHELL_SUBCMD_SET_END
+};
+
+SHELL_CREATE_STATIC_SUBCMD_SET(cfb_cmds) {
+	SHELL_CMD_ARG(init, NULL, HELP_NONE, cmd_init, 1, 0),
+	SHELL_CMD_ARG(get_device, NULL, HELP_NONE, cmd_get_device, 1, 0),
+	SHELL_CMD(get_param, &sub_cmd_get_param,
+		  "<all, height, width, ppt, rows, cols>", NULL),
+	SHELL_CMD_ARG(get_fonts, NULL, HELP_NONE, cmd_get_fonts, 1, 0),
+	SHELL_CMD_ARG(set_font, NULL, "<idx>", cmd_set_font, 2, 0),
+	SHELL_CMD_ARG(invert, NULL, HELP_NONE, cmd_invert, 1, 0),
+	SHELL_CMD_ARG(print, NULL, HELP_PRINT, cmd_print, 4, 0),
+	SHELL_CMD(scroll, &sub_cmd_scroll, "<dir: (vertical|horizontal)> "
```

Comment:
I think that you might consider changing help message a bit, it seems that it is duplicated.
Please call `scroll -h` to see what I mean. You can put here a simple description what `scroll` is resposible for, like scrolling what :).

Reply:
Addressed.